### ISSUE TITLE
/sendFile end point ignores 'filename' request parameter

### ISSUE THREAD

**Issue by ehussain** [allburov/whatsapp-http-api#133]
Hello,

I sent a request to /api/sendFile end point which contains following json.

```
(
    [json] => Array
        (
            [chatId] => xxxx@c.us
            [session] => 0c79e8f0-1bf0-11ee-aac6-4bba6cfec57f
            [type] => media
            [file] => Array
                (
                    [mimetype] => application/pdf
                    [filename] => 1688567559635.pdf
                    [url] => https://7c89-116-72-22-45.ngrok-free.app/file/r/387fc6a0-1ef5-11ee-a9e1-e12a60ff0642
                )

            [caption] => 
        )
)
```
note : this is php array structure but json request was successful. I got response and file on whatsapp as well.

However the whatsapp shows file with this information.

![image](https://github.com/devlikeapro/whatsapp-http-api/assets/358413/7a7171d7-a85d-4710-b9fb-d70907d8d956)

As you can see the caption and filename is not received correctly. 

Expected output.
File should be received with name as '1688567559635.pdf' and without any caption.

**Comment by allburov** [comment]
Hi! Could you attach JSON request that can be tested in swagger directly please? 
Just to make sure there's no language related serialization issue too

**Comment by ehussain** [comment]
Hello,

Attached the exact json request I am sending. I have masked the phone number.

```
{"chatId":"XXXXXXXXXXXX@c.us","session":"0c79e8f0-1bf0-11ee-aac6-4bba6cfec57f","type":"media","file":{"mimetype":"application\/pdf","filename":"1688567559635.pdf","url":"https:\/\/7c89-116-72-22-45.ngrok-free.app\/file\/r\/20391ec0-1f02-11ee-b611-23762cf3fa89"},"caption":null}
```

**Comment by allburov** [comment]
I've fixed the issue with `filename` field - it should work now in the latest version (just published it)

About `caption` -  he have a ticket for that already, sorry, didn't find it first time https://github.com/devlikeapro/whatsapp-http-api/issues/94
It works only in NOWEB engine. Just checked - it's not ready in default (WEBJS) one yet

Let me know if filename works for you and we can close the issue (we'll work on caption in the https://github.com/devlikeapro/whatsapp-http-api/issues/94)

**Comment by ehussain** [comment]
Hello. We are using 'WEBJS' engine right now. Is it available in that engine?

We have to use WEBJS to ensure that phone number doesn't get blocked.

**Comment by allburov** [comment]
Filename - yes, caption - not :(

**Comment by devlikepro** [comment]
Hi! 
`caption` has been fixed in **WAHA Plus** `2024.3.9`
Now both `caption` and `filename` should work with no issues in both WEBJS and NOWEB engine.

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
